core/version_test.go

```diff
@@ -42,3 +43,66 @@ func TestCannotParseBlockVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportedBlockVersion(t *testing.T) {
+	testVersions := []struct {
+		block  semver.Version
+		latest semver.Version
+	}{
+		{
+			block:  *core.LatestVer,
+			latest: *core.LatestVer,
+		},
+		{
+			block:  core.LatestVer.IncPatch(),
+			latest: *core.LatestVer,
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncPatch(),
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncMinor(),
+		},
+		{
+			block:  *core.LatestVer,
+			latest: core.LatestVer.IncMajor(),
+		},
+	}
+
+	latestVarTemp := core.LatestVer
+
+	for i, test := range testVersions {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			core.LatestVer = &test.latest
+			err := core.CheckBlockVersion(test.block.String())
+			assert.NoError(t, err)
+		})
+	}
+
+	core.LatestVer = latestVarTemp
+}
+
+func TestUnsupportedBlockVersion(t *testing.T) {
+	testVersions := []struct {
+		block  semver.Version
+		latest semver.Version
+	}{
+		{
+			block:  core.LatestVer.IncMajor(),
+			latest: *core.LatestVer,
+		},
+		{
+			block:  core.LatestVer.IncMinor(),
+			latest: *core.LatestVer,
+		},
```

Comment:
Add comments for `TestUnsupportedBlockVersion` test cases

```suggestion
		{
			// Block is newer major than latest (e.g., 1.0.0 > 0.14.0)
			block:  core.LatestVer.IncMajor(),
			latest: *core.LatestVer,
		},
		{
			// Block is newer minor than latest (e.g., 0.15.0 > 0.14.0)
			block:  core.LatestVer.IncMinor(),
			latest: *core.LatestVer,
		},
```